.github/workflows/llvmlite_win-64_wheel_builder.yml

```
@@ -148,5 +151,54 @@ jobs:
 
       - name: Install and test
         run: |
-          pip install dist/*.whl
+          # Upgrade pip and install wheel
+          python -m pip install --upgrade pip wheel
+
+          # Install wheel and run tests
+          cd dist
+          # Windows compatible way to get the wheel file using bash
+          whl=$(ls -1 ./*.whl)
+          echo "Using wheel: $whl"
+          python -m pip install "$whl"
+
+          # Run tests
           python -m llvmlite.tests
+
+  win-64-upload:
+    name: win-64-upload
+    needs: win-64-test
+    if: github.event_name == 'workflow_dispatch'
+    runs-on: windows-2019
+    defaults:
+      run:
+        shell: bash -elx {0}
+    strategy:
+      matrix:
+        python-version: ["3.10", "3.11", "3.12", "3.13"]
+      fail-fast: false
+    steps:
+      - name: Setup Miniconda
+        uses: conda-incubator/setup-miniconda@v3
+        with:
+          python-version: ${{ env.VALIDATION_PYTHON_VERSION }}
+          conda-remove-defaults: true
+          auto-update-conda: true
+          auto-activate-base: true
+
+      - name: Install anaconda-client
+        run: conda install -c anaconda anaconda-client
+
+      - name: Download llvmlite wheel
+        uses: actions/download-artifact@v4
+        with:
+          name: llvmlite-win-64-py${{ matrix.python-version }}
+          path: dist
+
+      - name: Upload wheel to Anaconda Cloud
+        env:
+          ANACONDA_API_TOKEN: ${{ secrets.NUMBA_CHANNEL_WHEEL_UPLOAD }}
+        run: |
+          cd dist
+          whl=$(stat -f "%m %N" ./*.whl | sort -n | tail -n 1 | cut -d' ' -f2-)
```

```suggestion
          whl=$(ls -1 ./*.whl)
```

As used above? Or does that not work in this context?

---

yes, this expression was from other workflow. stat command not on win, and ls is simpler.